### PR TITLE
Bug 1342191

### PR DESCRIPTION
**Pull request by tyagi-iiitv**
balrog agent fails to process subsequent changes if an exception is hit for an earlier one. r=bhearsum

### CODE REVIEW

File: agent/balrogagent/cmd.py [hunk 1]
```diff
@@ -62,11 +62,12 @@
                         resp.close()
                     else:
                         logging.debug("Change %s is not ready", change["sc_id"])
+                        raise
```
Comment (tyagi-iiitv) [inline agent/balrogagent/cmd.py]: I tried to raise a virtual exception for the test case here

**Comment by mozbhearsum** [comment]
It looks like you're still having a bit of trouble getting the tests set-up properly - is there anything I can do to help?

**Comment by tyagi-iiitv** [comment]
Ohh not really .. I am on a vacation and will be back next week. Is this urgent ?

**Comment by mozbhearsum** [comment]
Not at all, take your time!

**Comment by tyagi-iiitv** [comment]
Hi Ben, I actually tried testing this one, and based on the test, I've a few suggestions which I listed in the review comments. Please have a look at them and see if replacing **raise** with **pass** will work in this case. Because if we use **raise**, we can't continue with the loop execution after an exception is raised.

**Comment by mozbhearsum** [comment]
I had a reread of everything here and I'm not quite sure where you're getting stuck. Can you expand?

A couple of overall comments though:
- "raise_exceptions" should probably go away - raising them is incompatible with fixing this bug
- I think you're right that "pass" needs to be used in the "except" block, but make sure you still log the exceptions beforehand.
- The tests are going to the tricky part - you need a test that makes two requests, the first of which needs to cause an exception to be raised, the second of which should not.

**Comment by mozbhearsum** [comment]
Closing due to inactivity. Feel free to re-open if you'd like to finish this up.

**Pull request closed without merging**